### PR TITLE
MLflow fix for tests on Mac Python 3.9 and 3.10

### PR DESCRIPTION
**Pull request by strickvl**
## Describe changes
I fixed MlFlow example test failures on MacOS 3.9/3.10 by reworking the example and using the sklearn model instead of the TF one. The original root cause is not quite clear still, since there was no version difference in pips of working/non-working environments.

No bugs of MlFlow were identified in the core code, which is quite good anyway.

OSSK-41

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to docs, I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).
- [x] If my changes require changes to the dashboard, these changes are communicated/requested.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Other (add details above)

## Summary by CodeRabbit

- **New Features**
	- Transitioned machine learning integrations from TensorFlow to scikit-learn across various MLflow pipelines, enhancing compatibility and performance.
	- Introduced scikit-image for image processing requirements in machine learning operations.
	- Updated MLflow pipelines to support scikit-learn models, including changes in data preprocessing, model training, evaluation, and deployment processes.
	- Modified MLflow tracking and deployment configurations to align with scikit-learn integration.

- **Bug Fixes**
	- Corrected data normalization factors to ensure compatibility with scikit-learn's data processing requirements.
	- Removed unnecessary operations such as argmax from prediction steps to streamline model evaluation and prediction processes.

- **Refactor**
	- Updated import statements and internal functionalities to replace TensorFlow dependencies with scikit-learn and scikit-image across various steps in MLflow pipelines.
	- Renamed and modified functions and variables to reflect the shift to scikit-learn, improving code clarity and maintainability.

- **Tests**
	- Expanded test coverage to include the updated MLflow pipelines and steps, ensuring robustness and reliability of the new scikit-learn integrations.

### CODE REVIEW

**Comment by avishniakov** [comment]
You need to remove
```python
# TODO: remove this temporary disabling of the test for Python 3.9 and 3.10
    #  once the MLflow issue is resolved
    if sys.platform != "darwin" and sys.version_info[:2] not in [
        (3, 9),
        (3, 10),
    ]:
```
to test this.
Also reducing what we call tests in `test-coverage-xml.sh` might be handy for testing

**Comment by avishniakov** [comment]
@coderabbitai review

File: tests/integration/examples/mlflow/steps/trainer_step.py [hunk 1]
```diff
@@ -36,29 +36,15 @@
 def trainer(
     X_train: np.ndarray,
     y_train: np.ndarray,
-    epochs: int = 5,
-    lr: float = 0.001,
-) -> tf.keras.Model:
-    """Train a neural net from scratch to recognize MNIST digits return our
+) -> SVC:
+    """Train a RF Classifier to recognize MNIST digits return our
```
Comment (strickvl) [inline tests/integration/examples/mlflow/steps/trainer_step.py:40]:
```suggestion
    """Train an SVC Classifier to recognize MNIST digits return our
```

**Review by strickvl — COMMENTED** [review]
Just one tiny nit, but otherwise looks good to me.
